porcupine/plugins/blockcursor.py

```diff
@@ -0,0 +1,27 @@
+"""Toggles between the ❚ and | cursor state of the textwidgets."""
+from __future__ import annotations
+
+import tkinter
+
+from porcupine import get_tab_manager, settings, tabs
+
+
+def do_toggle(event: tkinter.Event[tkinter.Text]) -> None:
+    if settings.get("blockcursor", object):
+        event.widget.configure(blockcursor=True, insertwidth=0)  # minimize the cursor thickness
+    else:
+        event.widget.configure(
+            blockcursor=False, insertwidth=2
+        )  # set insertwidth back to tk default
+
+
+def on_new_filetab(tab: tabs.FileTab) -> None:
+    is_block = settings.get("blockcursor", object)
+    tab.textwidget.config(blockcursor=is_block, insertwidth=0 if is_block else 2)
```

Comment:
To avoid fragile copy/pasta, it would be better to do something like this:

```suggestion
    do_toggle(tab)
```

Then `do_toggle()` has to be changed so that it accepts a text widget, not an event. When binding, you can use e.g. `lambda event: do_toggle(tab)`.

Reply:
Thanks, it's much better indeed.